crates/adapters/src/integrated/delta_table/input.rs

```diff
@@ -322,12 +322,21 @@ impl DeltaTableInputEndpointInner {
     /// table declaration.
     fn used_columns(&self, table: &DeltaTable) -> Vec<String> {
         // Column names in the SQL schema.
-        let sql_columns = self
-            .schema
-            .fields
-            .iter()
-            .map(|field| field.name.name())
-            .collect::<Vec<String>>();
+        let sql_columns = if self.config.skip_unused_columns {
+            self.schema
+                .fields
+                .iter()
+                // skip unused columns as long as they are nullable or have a default value.
+                .filter(|f| !f.unused || (!f.columntype.nullable && f.default.is_none()))
```

Comment:
With some more work in the compiler I think we could skip all of them.
We should just not synthesize them in the dataflow graph.
But this would need to be a compiler flag.